Quotient/blurhash.h

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2024 Joshua Goins <josh@redstrate.com>
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "quotient_export.h"
+
+#include <QImage>
+
+class TestBlurHash;
+
+namespace Quotient {
+/**
+ * @brief Encodes and decodes image to and from the BlurHash format. See https://blurha.sh/.
+ *
+ * @note This class has been adapted from https://github.com/redstrate/QtBlurHash.
+ */
+class QUOTIENT_API BlurHash
+{
+public:
+    /** Decodes the @p blurhash string creating an image of @p size.
+     * @note Returns a null image if decoding failed.
+     */
+    static QUOTIENT_API QImage decode(const QString &blurhash, const QSize &size);
+
+    /** Encodes the @p image and returns a blurhash string.
+     * @param image A non-null image.
+     * @param componentsX the number of components X-wise. Must be between 1 and 9.
+     * @param componentsY the number of components Y-wise. Must be between 1 and 9.
+     * @note Returns an empty string if it failed to encode the image.
+     */
+    static QUOTIENT_API QString encode(const QImage &image, int componentsX = 4, int componentsY = 4);
+
+protected:
```

Comment:
`protected` implies inheriting from a class - do you expect that clients should derive from this class and somehow use its static internals?